src/sirfilesystem.c

```diff
@@ -46,7 +46,11 @@ bool _sir_pathgetstat(const char* restrict path, struct stat* restrict st, sir_r
     if (relative) {
 #if !defined(__WIN__)
 # if defined(__MACOS__) || defined(_AIX)
+#  if !defined(O_SEARCH)
```

Comment:
O_SEARCH should always be defined if _DARWIN_C_SOURCE is, but I have seen this happen on Traivs machines. I have looked at the headers, and I don't understand it.

Have you confirmed that it still works with O_DIRECTORY?

Reply:
Actually, it's more nuanced. There is a selector for _DARWIN_C_LEVEL in sys/cdefs.h I believe. It has _ANSI_SOURCE and _POSIX_C_SOURCE involved.

Reply:
And _DARWIN_C_LEVEL has to be _DARWIN_C_FULL

Reply:
Yeah, this is ONLY if O_SEARCH isn't defined, and that's going to happen only on some older Xcode releases / they seem to have added it sometime around Xcode 12.  I verified that O_DIRECTORY does indeed work and pass all tests if O_SEARCH isn't available.